nservicebus/outbox/index.md

```diff
@@ -79,8 +79,64 @@ In order to gradually convert an entire system from the DTC to the Outbox:
 WARNING: When verifying outbox functionality, it can sometimes be helpful to temporarily [stop the MSDTC Service](https://technet.microsoft.com/en-us/library/cc770732.aspx). This ensures that the Outbox is working as expected, and no other resources are enlisting in distributed transactions.
 
 
+## Message identity and idempotent processing
+
+If handlers are idempotent then outbox might not be required at all. Outbox relies on message identity to guarantee consistency. When using outbox on both the sender and receiver message identifiers are consistent if a handler is invoked more-than-once for a message. 
+If a sender does not use the outbox or when sending a message outside of a handler the sender *must* generate and apply the same message identifier to guarantee consistency at the receiver.
+The same message identity is also important to have deduplication working on the infrastructure level on some transport that support this. [Read more on how to set a message identifier.](/nservicebus/messaging/message-identity.md)
+
+
+## Concurrency
+
+Outbox cannot guarantee that handlers or sagas are invoked exactly once. Outbox does not prevent concurrent attempts to process the same message but ensures that the outcome of the processing is persisted only once for resources that share the outbox transaction.
```

Comment:
In the end, this sentence just confirms the outbox limitations of only providing "exactly-once" semantics for the outbox storage + transport. All other resources continue to be exposed to duplicate messages. I'd rather ensure to clarify this concept and avoid this paragraph all together.